src/sidecar/connections/watcher.ts

```diff
@@ -41,17 +47,42 @@ export async function waitForConnectionToBeStable(
     // and trigger any kind of Topics/Schemas refresh to reenforce the error state / clear out any
     // old data
     environmentChanged.fire(id as unknown as EnvironmentId);
+    const lastConnectionEvent = connectionStateWatcher.getLatestConnectionEvent(id);
+    if (lastConnectionEvent) {
+      showErrorNotificationWithButtons(
+        `Timed out establishing "${lastConnectionEvent.connection.spec.name}" connection.`,
+      );
+    }
     logger.error(msg);
     return null;
   }
 
   logger.debug(
     `waitForConnectionToBeStable(): connection is stable after ${Date.now() - startTime}ms, returning`,
   );
+  notifyForFailedState(connection);
 
   return connection;
 }
 
+function notifyForFailedState(connection: Connection) {
+  let failedStatuses: string[] = [];
+  // the notifications don't allow rich formatting here, so we'll just list out the failed resources
+  // and then try to show the errors themselves in the item tooltips
+  if (connection.status.ccloud?.state === "FAILED") failedStatuses.push("Confluent Cloud");
+  if (connection.status.kafka_cluster?.state === "FAILED") failedStatuses.push("Kafka");
+  if (connection.status.schema_registry?.state === "FAILED") failedStatuses.push("Schema Registry");
+  if (failedStatuses.length > 0) {
+    showErrorNotificationWithButtons(
+      `Failed to establish connection to ${failedStatuses.join(" and ")} for "${connection.spec.name}".`,
+      {
+        "View Connection Details": () =>
+          commands.executeCommand(`confluent.connections.direct.edit`, connection.id),
```

Comment:
Seems wrong to invite if the connection in question is ccloud. waitForConnectionToBeStable() is called with ccloud  as well as any direct connection (at least if its docstring is still correct).

Need separate handling entirely if connection is ccloud vs direct?

Reply:
Good catch; done in https://github.com/confluentinc/vscode/pull/946/commits/608e3b72c005703002e57f01681a7be15710e324